SEED-platform/seed
# 239 progress bar

#### Any background context you want to provide?

The progress bar code was difficult to manage due to the various methods that needed to be called in the correct order. For example:

```
prog_key = get_prog_key('check_data', identifier)

data_quality_results = get_cache_raw(DataQualityCheck.cache_key(identifier))
result = {
	'status': 'success',
	'progress': 100,
	'progress_key': prog_key,
    'message': 'data quality check complete',
    'data': data_quality_result
}
set_cache(prog_key, result['status'], result)
return result
```

now becomes

```
cache_key, dq_id = DataQualityCheck.initialize_cache(import_file_id)
progress_data = ProgressData(func_name='check_data', unique_id=dq_id)
return progress_data.result()
```

There is also the ability to initialize the ProgressData object from an existing key (in the redis db) by calling `progress_data = ProgressData.from_key(progress_key)`

The ability to save the results of any progress is much easer now too by calling `progress_data.finish_with_error('StopIteration Exception', traceback.format_exc())`. There are various helper methods that set all the required status, progress, etc fields.

#### What's this PR do?

* General
	* Upgrade celery to 4.2. Required updating tasks that called sub-tasks. 
	* Remove scipy, ipython, and numpy from local.txt requirements. These are now in local-scipy.txt if needed (e.g. for manual migrations)
	* Added new ProgressData class to manage tracking the progress of background tasks
	* Update test_seed_host_api.py (seed_readingtools.py) to support secondary call to data_quality_checks.
* Matching
	* Move equivalence partitioner code into its own file
* MCM
	* Removed old FakeModel, FakeManager, etc code
* Testing
	* New FakePropertyStateFactory method for returning data in the extra_data field to be used for testing raw_data import and mapping
	* Break up the data_importer tests into integration, views, and unit tests. Integration tests require the loading of CSV/XLSX files
* API
	* Removed endpoint `/api/v2/import_files/{id}/data_quality_results`. Use `/api/v2/data_quality_checks/results/?organization={org_id}&data_quality_id={dq_id}`
	* Add endpoint for starting data quality on import file. `/api/v2/import_files/{id}/start_data_quality_checks/?organization={org_id}`
	* Progress endpoints/results no longer return data such as the number of records matched or the data quality results. You must request those from subsequent result APIs.


#### TODOs

* There are several TODO 239 comments that need to be addressed by @nllong
* @axelstudios - can you check the export data from the inventory page. The modal dialog doesn't appear anymore.

#### How should this be manually tested?

* Import files with and without always eager
* Verify data quality works on both mapping and within inventory
* Verify data quality export works in both contexts
* Verify mapping and matching
* Verify deleting inventory and deleting organization with and without always eager

#### What are the relevant tickets?

#239 
#1667
#873 - Is no longer applicable, or at least shouldn't happen again.
Working towards #1498 and #1608

#### Screenshots (if appropriate)
#### Definition of Done:
